src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests_DefaultsSource.vb

```diff
@@ -186,14 +184,112 @@ namespace NS1
         Private Class MockDefaultSource
             Implements IAsyncCompletionDefaultsSource
 
+            Public Shared Defaults As ImmutableArray(Of String) = ImmutableArray(Of String).Empty
+
             <ComponentModel.Composition.ImportingConstructor>
             <Obsolete(MefConstruction.ImportingConstructorMessage, True)>
             Public Sub New()
             End Sub
 
             Public Function GetSessionDefaultsAsync(session As IAsyncCompletionSession) As Task(Of ImmutableArray(Of String)) Implements IAsyncCompletionDefaultsSource.GetSessionDefaultsAsync
-                Return Task.FromResult(ImmutableArray.Create("MyAB", "MyA"))
+                Return Task.FromResult(Defaults)
+            End Function
+        End Class
+
+        <WpfFact>
+        Public Async Function SelectDefaultItemOverStarredItem() As Task
+            Using state = TestStateFactory.CreateCSharpTestState(
+                <Document>
+class MyClass 
+{                          
+    public bool FirstStarred()  => true;
+    public bool SecondStarred() => true;
+    public bool FirstDefault()  => true;
+}
+class Test
+{
+    void M(MyClass c)
+    {
+        c$$
+    }
+}
+                </Document>,
+                extraExportedTypes:={GetType(IntelliCodeMockProvider), GetType(MockDefaultSource)}.ToList())
+
+                MockDefaultSource.Defaults = ImmutableArray.Create("FirstDefault")
```

Comment:
I don't fully understand what "default" means here. Do you mind explaining a bit?

Reply:
The purpose of "default" is to make the selection of completion item consistent with the grey text provided by IntelliCode's wholeline completion.

This is the original PR for this feature, I can go into more details if you want, just let me know :)
https://github.com/dotnet/roslyn/pull/57685

Reply:
it might be worth doc'ing the test.  i.e. that we are still starring SecondStarred here even though the IntelliCodeMockProvider doesn't return it.

--

NOte: does anything confirm that the starred item appears at the top above hte other items? like does AssertCompletionItemsContainAll asssert that those are in the right order?

Reply:
Sure. Will do

For the order of items, starred items are definitely NOT at the top unless user hasn't typed anything yet. We currently sorted items by pattern matching. There has been discussion before on whether to keep them at the top always, and I even have a [PR ](https://github.com/dotnet/roslyn/pull/65199) for that. But I'm not sure we reached a conclusion on the desired behavior yet.

Reply:
> i.e. that we are still starring SecondStarred here even though the IntelliCodeMockProvider doesn't return it.

oh, I might have misunderstood your comment. This change doesn't add star to default item. It only fixes the selection (select default over starred item.)

IntelliCode provider does some additional customizations on starred items (like sort text), I'd like to extract and share that logic (as well as some general clean up around stared item handling in Roslyn) in a follow up.